src/Umbraco.Core/IO/FileSystems.cs

```diff
@@ -38,92 +40,137 @@ public class FileSystems : IFileSystems
         #region Constructor
 
         // DI wants a public ctor
-        public FileSystems(IServiceProvider container, ILogger<FileSystems> logger, ILoggerFactory loggerFactory, IIOHelper ioHelper, IOptions<GlobalSettings> globalSettings, IHostingEnvironment hostingEnvironment)
+        public FileSystems(
+            ILoggerFactory loggerFactory,
+            IIOHelper ioHelper,
+            IOptions<GlobalSettings> globalSettings,
+            IHostingEnvironment hostingEnvironment,
+            IFileSystem stylesheetFileSystem)
```

Comment:
If we are going to allow injection of the stylesheetFileSystem, it must be `IStylesheetFileSystem` which inherits from `IFileSystem`. Each one must be a unique service because MSDI doesn't support named services.

There was previous issues when trying to inject the media file system, so just wondering why this isn't a problem here?

Reply:
Me saying that we inject is a bit misleading, I don't actually register the filesystem in the container and inject it, I pass it to the constructor using `provider.CreateInstance<FileSystems>(stylesheetFileSystem);` that way the filesystem never actually gets registered in the container, it just gets passed to the constructor when FileSystems is created. 

The reason that the media file system was an issue was because we needed to shadow the inner media file system, and not the media file system itself, but the stylesheet filesystem is just an arbitrary IFileSystem that's only accessed through FileSystems, so that's why it's not an issue, but I think it's better to go with Rasmus suggestion, that way we don't need to pass anything to the constructor.